File: controledeestoque/src/main/java/com/inventario/imobilizado/model/Item.java
package com.inventario.imobilizado.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;


@Entity
@Table(name = "item")
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id_item")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id_item;

    private String descricao;

    private String localizacao_atual;

    private Integer potencia;

    private String patrimonio;

    private String numero_de_serie;

    private String numero_nota_fiscal;

    private String comentario_manutencao;

    @Temporal(TemporalType.DATE)
    private Date data_entrada;

    @Temporal(TemporalType.DATE)
    private Date ultima_qualificacao;

    @Temporal(TemporalType.DATE)
    private Date data_nota_fiscal;

    @Temporal(TemporalType.DATE)
    private Date proxima_qualificacao;

    @Temporal(TemporalType.DATE)
    private Date prazo_manutencao;

    private String estado;

    @ManyToOne
    @JoinColumn(name = "categoria_id_categoria")
    private Category categoria;

    @Column(name = "status_item")
    private String status;

    @ManyToOne
    @JoinColumn(name = "modelo_id_modelo")
    private Modelo modelo;

    @ManyToOne
    @JoinColumn(name = "localizacao_id_localizacao")
    private Location localizacao;

}

File: projeto_integrador/lib/models/item.dart
class Item {
  final int id;
  final String descricao;
  final String localizacaoAtual;
  final int? potencia;
  final String estado;
  final String numeroDeSerie;
  final String numeroNotaFiscal;
  final String comentarioManutencao;
  final String patrimonio;
  final int categoria;
  final String status;
  final int modelo;
  final String marca;
  final int localizacao;
  final DateTime dataEntrada;
  final DateTime ultimaQualificacao;
  final DateTime dataNotaFiscal;
  final DateTime proximaQualificacao;
  final DateTime prazoManutencao;

  Item({
    required this.id,
    required this.descricao,
    required this.localizacaoAtual,
    required this.potencia,
    required this.estado,
    required this.numeroDeSerie,
    required this.numeroNotaFiscal,
    required this.comentarioManutencao,
    required this.patrimonio,
    required this.categoria,
    required this.status,
    required this.modelo,
    required this.marca,
    required this.localizacao,
    required this.dataEntrada,
    required this.ultimaQualificacao,
    required this.dataNotaFiscal,
    required this.proximaQualificacao,
    required this.prazoManutencao,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id_item'],
      descricao: json['descricao'],
      localizacaoAtual: json['localizacao_atual'],
      potencia: json['potencia'],
      estado: json['estado'],
      numeroDeSerie: json['numero_de_serie'],
      numeroNotaFiscal: json['numero_nota_fiscal'],
      comentarioManutencao: json['comentario_manutencao'],
      patrimonio: json['patrimonio'],
      categoria: json['categoria'],
      status: json['status'],
      modelo: json['modelo'],
      marca: json['marca'],
      localizacao: json['localizacao'],
      dataEntrada: DateTime.parse(json['data_entrada']),
      ultimaQualificacao: DateTime.parse(json['ultima_qualificacao']),
      dataNotaFiscal: DateTime.parse(json['data_nota_fiscal']),
      proximaQualificacao: DateTime.parse(json['proxima_qualificacao']),
      prazoManutencao: DateTime.parse(json['prazo_manutencao']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_item': id,
      'descricao': descricao,
      'localizacao_atual': localizacaoAtual,
      'potencia': potencia,
      'estado': estado,
      'numero_de_serie': numeroDeSerie,
      'numero_nota_fiscal': numeroNotaFiscal,
      'comentario_manutencao': comentarioManutencao,
      'patrimonio': patrimonio,
      'categoria': categoria,
      'status': status,
      'modelo': modelo,
      'marca': marca,
      'localizacao': localizacao,
      'data_entrada': dataEntrada.toIso8601String(),
      'ultima_qualificacao': ultimaQualificacao.toIso8601String(),
      'data_nota_fiscal': dataNotaFiscal.toIso8601String(),
      'proxima_qualificacao': proximaQualificacao.toIso8601String(),
      'prazo_manutencao': prazoManutencao.toIso8601String(),
    };
  }
}
